packages/core/src/integrations/supabase.ts

```diff
@@ -0,0 +1,488 @@
+
+              const data: Record<string, unknown> = {};
+
+              if (query.length) {
+                data.query = query;
+              }
+
+              if (Object.keys(body).length) {
+                data.body = body;
+              }
+
+              if (Object.keys(data).length) {
+                breadcrumb.data = data;
+              }
+
+              addBreadcrumb(breadcrumb);
+
+              return res;
+            },
+            (err: Error) => {
+              if (span) {
+                setHttpStatus(span, 500);
+                span.end();
+              }
+              throw err;
+            },
+          )
+          .then(...argumentsList);
+      },
+    },
+  );
+}
+
+function instrumentPostgrestQueryBuilder(PostgrestQueryBuilder: new () => PostgrestQueryBuilder): void {
+  if (instrumented.has(PostgrestQueryBuilder)) {
+    return;
+  }
+
+  // We need to wrap _all_ operations despite them sharing the same `PostgrestFilterBuilder`
+  // constructor, as we don't know which method will be called first, an we don't want to miss any calls.
+  for (const operation of AVAILABLE_OPERATIONS) {
+    instrumented.set(PostgrestQueryBuilder, {
+      [operation]: (PostgrestQueryBuilder.prototype as Record<string, unknown>)[
+        operation as 'select' | 'insert' | 'upsert' | 'update' | 'delete'
+      ] as (...args: unknown[]) => PostgrestFilterBuilder,
+    });
+
+    type PostgrestOperation = keyof Pick<PostgrestQueryBuilder, 'select' | 'insert' | 'upsert' | 'update' | 'delete'>;
+    (PostgrestQueryBuilder.prototype as Record<string, any>)[operation as PostgrestOperation] = new Proxy(
+      (PostgrestQueryBuilder.prototype as Record<string, any>)[operation as PostgrestOperation],
+      {
+        apply(target, thisArg, argumentsList) {
+          const rv = Reflect.apply(target, thisArg, argumentsList);
+          const PostgrestFilterBuilder = (rv as PostgrestFilterBuilder).constructor;
+
+          logger.log(`Instrumenting ${operation} operation's PostgrestFilterBuilder`);
+
+          instrumentPostgrestFilterBuilder(PostgrestFilterBuilder);
+
+          return rv;
+        },
+      },
+    );
+  }
+}
+
+const instrumentSupabase = (supabaseClientInstance: unknown): void => {
+  if (!supabaseClientInstance) {
+    throw new Error('Supabase client instance is not available.');
+  }
+  const SupabaseClientConstructor =
+    supabaseClientInstance.constructor === Function ? supabaseClientInstance : supabaseClientInstance.constructor;
+
+  instrumentSupabaseClientConstructor(SupabaseClientConstructor);
+  instrumentSupabaseAuthClient(supabaseClientInstance as SupabaseClientInstance);
+};
+
+const INTEGRATION_NAME = 'Supabase';
+
+const _supabaseIntegration = (supabaseClient => {
+  return {
+    name: INTEGRATION_NAME,
+    setupOnce() {
+      instrumentSupabase(supabaseClient);
+    },
+  };
+}) satisfies IntegrationFn;
+
+export const supabaseIntegration = defineIntegration((supabaseClient: unknown) => {
```

Comment:
Should the `supabaseIntegration` be added by default? If yes, then you should add it to `getDefaultIntegrations`.

Edit: Ah, as the `supabase` client needs to be passed, it actually cannot be default 🤔